mkosi/__init__.py

```diff
@@ -2067,7 +2067,8 @@ def python_binary(config: Config, *, binary: Optional[PathString]) -> str:
 
     # If there's no tools tree, prefer the interpreter from MKOSI_INTERPRETER. If there is a tools
     # tree, just use the default python3 interpreter.
-    return "python3" if tools and config.tools_tree else sys.executable
+    exe = Path(sys.executable)
+    return "python3" if (tools and config.tools_tree) or not exe.is_relative_to("/usr") else exe
```

Comment:
```suggestion
    exe = sys.executable
    return "python3" if (tools and config.tools_tree) or not Path(exe).is_relative_to("/usr") else exe
```
and we can keep the return value a string or
```suggestion
    exe = Path(sys.executable)
    return Path("python3") if (tools and config.tools_tree) or not exe.is_relative_to("/usr") else exe
```
and make the return value a `Path`. I'd prefer the former.

Reply:
I dunno PathString seems to convey best here what we want.